cranelift/isle/isle/src/parser.rs

```diff
@@ -12,12 +12,21 @@ pub fn parse(lexer: Lexer) -> Result<Vec<Def>> {
     parser.parse_defs()
 }
 
+/// Parse without positional information. Provided mainly to support testing, to
+/// enable equality testing on structure alone.
+pub fn parse_without_pos(lexer: Lexer) -> Result<Vec<Def>> {
+    let mut parser = Parser::new(lexer);
+    parser.disable_pos();
+    parser.parse_defs()
+}
```

Comment:
Previous discussion still applies: https://github.com/bytecodealliance/wasmtime/pull/8263#discussion_r1543966705. I still tend to think this is the simplest of the options.

@cfallin regarding the suggestion "a helper that maps through all AST nodes and sets positions to e.g. zero?". I feel like this could be quite a bit of boilerplate to implement. Happy to do that if you prefer.